### ISSUE TITLE
Central area datastore repository

### ISSUE THREAD

**Issue by dflynn-volpe** [VisionEval/VisionEval-Dev#16]
Originally created by @gregorbj in 2017

We need an efficient way to store and access a test area datastore. A test area datastore is essential to efficient development and testing of modules since most modules need to use data that is in a datastore. Currently I've been including two datastores in the 'test' directory of each package. In an 'initial_datastore' directory is a datastore with initial conditions. In the main directory is a datastore with the final conditions (after all package modules are run). The final datastore is then copied to the 'initial_datastore' folder for the next package being worked on. This results in a lot of duplicated data. It would take less space and perhaps be more efficient and less error-prone to read in the datastore.h5 file from the 'VERSPM' directory. This should be updated with the latest successful run of VERSPM. Might be good to have some functions to facilitate this and using the datastore in module testing.

This would require keeping a copy of the 'datastore.h5' file in the directory which we don't do now. It might also be good to keep a copy of the 'ModelState.Rda' file. This file makes it easier to read from the datastore. Having the test datastore would also be good for testing output and visualization functions and scripts.

Follow up Nov 2018:

I created a centralized datastore for VE-State test data in the sources/modules/Test_Data directory. The main reason for doing this now is that there is a lot of redundancy in storing test datastores in each of the module packages. While this isn't a problem for small test areas, it becomes a big problem for a large model area. I've altered the module test code in the VE-State modules so that it reads from and writes to this centralized datastore. It is a simple thing to do. This also advances the project towards the goal of creating centralized datastores that can be used for new module development and for regression testing. It would be simple to convert VE-RSPM and VE-RPAT datastores to centralized repositories as well since only some small changes to the test scripts are needed.

It would be a good idea to keep copies of all the def and input files for the test dataset in the same area to ensure replicability.

Follow up Feb 2019:
A trimmed down test will be needed for VE-State. The current test using Oregon data for 2010 and 2040 produces a datastore that is too big for GitHub.

**Comment by jrawbits** [comment]
I don't think we need a centralized datastore for models: we just need built-in working samples that generate "correct" results. So I'd rather keep things (like the sample model variants in VEModel) that are guaranteed to rebuild correctly, and then let people develop off that basis.

Users can change them by copying the model (the development-next VEModel makes that trivial). VEModel also makes it easy to base a model on another model (and dip into or copy its Datastore).

The difficulty with having the test data fully pre-built is that we will inevitably be doing things "deep enough" that the test data itself falls out of date - and we can't easily tell if the problem is the new development or failure to maintain the old.

**Comment by jrawbits** [comment]
This may also interact with #60 (testing protocol). If we do the VETest module as described in that other issue, part of what we can find there are models that will regenerate Datastores that are known to be correct.

**Comment by jrawbits** [comment]
There is an emerging simple test architecture underway in VE 3.x that provides a ready approach to work around this problem (allowing regeneration at will of partial or complete Datastores that can then be accessed by module tests). Review with the R 4.2.2 release to provide some advanced walkthrough examples of using an existing Datastore to test a new module using the "LoadModel" directive and a one-line run-model script to invoke the new module (mini-test model).

**Comment by jrawbits** [comment]
Closing this old issue since the installModel architecture makes it easy to construct test models. We still need a reduced model for VE-State testing, but that is (literally) a separate issue #61.
